src/pages/ThorChainLP/components/AddLiquidity/AddLiquidityInput.tsx

```diff
@@ -556,7 +556,10 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
       assetId: poolAsset?.assetId,
       spender: poolAssetInboundAddress,
       from: poolAssetAccountAddress,
-      amount: toBaseUnit(actualAssetDepositAmountCryptoPrecision, poolAsset?.precision ?? 0),
+      amountCryptoBaseUnit: toBaseUnit(
+        actualAssetDepositAmountCryptoPrecision,
+        poolAsset?.precision ?? 0,
```

Comment:
I still feel that a missing value here is super risky and should never default to 0. If anything we make precision optional and assert its presence at runtime?

Reply:
I do agree we're missing type-safety here which is definitely rank!

@woodenfurniture by making it optional do you mean in `toBaseUnit` / `fromBaseUnit` ? While I conceptually agree, feels like a *lot* of places would throw on first renders undefined precision and wouldn't want to risk it.

As a more local solution, I'd have

```typescript
  const actualAssetDepositAmountCryptoBaseUnit = useMemo(
    () =>
      poolAsset
        ? toBaseUnit(actualAssetDepositAmountCryptoPrecision, poolAsset?.precision ?? 0)
        : undefined,
    [actualAssetDepositAmountCryptoPrecision, poolAsset],
  )
```

But I'll hold off from it since this isn't the domain this PR touches - not a big deal to do small type improvements in other domains usually but `<AddLiquidityInput />` is already a monstrocity and has a dedicated PR to unmonster it https://github.com/shapeshift/web/issues/6996